src/Areas/ApplicationInsights/Commands/ListInsightsCommand.cs

```diff
@@ -0,0 +1,72 @@
+using AzureMcp.Areas.ApplicationInsights.Options;
+using AzureMcp.Areas.ApplicationInsights.Services;
+using AzureMcp.Services.Telemetry;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Areas.ApplicationInsights.Commands;
+
+public class ListInsightsCommand(ILogger<ListInsightsCommand> logger) : BaseProfilerCommand(logger)
+{
+    private const string CommandTitle = "List Code Optimization Insights";
+    private readonly Option<Guid> _appIdOption = new(
+        "--app-id",
```

Comment:
Keep consistent with the way the app commands use resource ID. Let's lookup app Id if we have to but don't require it or surface to customer if possible.

Reply:
Update in the new PR.